src/template/templateOptions.ts

```diff
@@ -14,11 +14,17 @@ export enum FileType {
     Index = "index"
 }
 
+export enum FileExtension {
+    Tsx = "withX",
+    Ts = "withoutX",
+}
```

Comment:
```suggestion
export enum FileExtensionType {
    WithX = "withX",
    WithoutX = "withoutX",
}
```
Small nitpick: `FileExtension` implies to me it's the values of file extensions, e.g `FileExtension.Tsx` = `.tsx`

Reply:
Done!